File: api/models/Word.js
/**
 * Word model
 * Represents a word with its definition, type, example and language
 */
class Word {
  constructor(word, type, definition, example, isFavorite = false, language = 'fr') {
    this.word = word;
    this.type = type;
    this.definition = definition;
    this.example = example;
    this.isFavorite = isFavorite;
    this.language = language;
  }

  toJSON() {
    return {
      word: this.word,
      type: this.type,
      definition: this.definition,
      example: this.example,
      isFavorite: this.isFavorite,
      language: this.language
    };
  }

  static fromJSON(json) {
    return new Word(
      json.word,
      json.type,
      json.definition,
      json.example,
      json.isFavorite || false,
      json.language || 'fr'
    );
  }
}

module.exports = Word;

File: app/lib/features/word_of_the_day/domain/models/word.dart
class Word {
  final String word;
  final String type;
  final String definition;
  final String example;
  final String date;
  bool isFavorite;
  final String language;

  Word({
    required this.word,
    required this.type,
    required this.definition,
    required this.example,
    required this.date,
    this.isFavorite = false,
    this.language = 'fr',
  });

  factory Word.fromJson(Map<String, dynamic> json) {
    return Word(
      word: json['word'],
      type: json['type'],
      definition: json['definition'],
      example: json['example'],
      date: json['date'],
      isFavorite: json['isFavorite'] ?? false,
      language: json['language'] ?? 'fr',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'word': word,
      'type': type,
      'definition': definition,
      'example': example,
      'date': date,
      'isFavorite': isFavorite,
      'language': language,
    };
  }
}
